crates/sui-core/src/consensus_adapter.rs

```diff
@@ -197,39 +229,20 @@ pub trait SubmitToConsensus: Sync + Send + 'static {
     ) -> SuiResult;
 }
 
+#[mockall::automock]
 #[async_trait::async_trait]
-impl SubmitToConsensus for TransactionsClient<sui_network::tonic::transport::Channel> {
-    async fn submit_to_consensus(
+pub trait ConsensusClient: Sync + Send + 'static {
+    async fn submit(
         &self,
         transactions: &[ConsensusTransaction],
-        _epoch_store: &Arc<AuthorityPerEpochStore>,
-    ) -> SuiResult {
-        let transactions_bytes = transactions
-            .iter()
-            .map(|t| {
-                let serialized =
-                    bcs::to_bytes(t).expect("Serializing consensus transaction cannot fail");
-                Bytes::from(serialized)
-            })
-            .collect::<Vec<_>>();
-        self.clone()
-            .submit_transaction(TransactionProto {
-                transactions: transactions_bytes,
-            })
-            .await
-            .map_err(|e| SuiError::ConsensusConnectionBroken(format!("{:?}", e)))
-            .tap_err(|r| {
-                // Will be logged by caller as well.
-                warn!("Submit transaction failed with: {:?}", r);
-            })?;
-        Ok(())
-    }
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> SuiResult<SubmitResponse>;
```

Comment:
The `SubmitToConsensus` trait didn't make sense anymore to re-use for the actual consensus client implementations. I've introduced the `ConsensusClient` which is now more focused on the actual consensus client .